### ISSUE TITLE
Random Error that Hit Server FPS Hard

### ISSUE THREAD

**Issue by ReDBaroN1** [MezoPlays/Exile-MostWanted#2]
Hey guys, just wanted to draw your attention to an issue that, when it happens, can hit the server FPS hard by spamming the rpt. Apologies but, I haven't been able to recreate the issue so, I don't know how it happens. But, when it happens the bounty row looks like this:
`76561198027960659	Marse	["N/A",scalar NaN,"Copic","76561198297517948"]	0	[]	[]	["76561198120894674","76561198027960659","76561198071643042","","76561197983987401","76561198155467585"]	27/06/2017 19:36	0000-00-00 00:00:00`
The part that is causing the trouble is the 'scalar NaN' in the middle of the bounty column. If I manually change that value or delete the row, it fixes it.

I really love this script so, it would be really superb if you could help me fix it?

**Comment by Mezo** [comment]
It's caused by the UI shitting itself and not passing the price of the bounty correctly. 

Personally, I have no interest in fixing this... and I very much doubt Wolf does. Sorry.

**Comment by ReDBaroN1** [comment]
aww dude! That's a real shame as it's one tiny little issue that's sort of become a show stopper. If left as is my only option is to pull it out as the spam dropping the FPS so much prevents missions from spawning. Plus, I don't know it's doing it until a player reports it....
But that would be a massive shame to pull it as it's such a superb idea and so well thought through.
Is there anything I can try. It's odd that it only seems to do this on my Malden server...Tanoa and Altis are fine but, then, Malden is the more populated....
Anyway, thanks for your response.

**Comment by itsthedevman** [comment]
I have an idea of how to attempt to fix it but as for testing, I have no ability to be able to do so. I won't be able to write up something until later today since currently I have no computer. I'll let you know

**Comment by ReDBaroN1** [comment]
Oh, that's superb news! Thanks a lot. 

I can test any fixes as I'm working from home tomorrow. However, I haven't been able to recreate the issue anyway but, am happy to run the fix on live and see what happens. 

Thanks again.

**Comment by itsthedevman** [comment]
It's hard to say for sure what is the cause of the issue and impossible to replicate in testing. 
Do you notice the person who sets the bounty respect getting messed up? Like scalar NaN, or being increased exponentially?

The first thing I would try replacing is this: 
Line 14 of ExileClient_MostWanted_Dialog_addBounty.sqf
Replace 
`_value = (lbData [2101, (lbCurSel 2101)]);`
With
`_value = (_pricelist lbData (lbCurSel _pricelist));`

Also, worthwhile to add a logging message in ExileServer_MostWanted_bounty_network_addBounty.sqf. Around line 84, you will have this line:
`["newBountyNotification",[MostWanted_MasterBountyList,str(_moneyAmount)]] call ExileServer_system_network_send_broadcast;`
Add this after it: 
`format["Player %1 (%2) has had a bounty set on them by %3 (%4). Price: %5 | Respect: %6 | DEBUG: %7", name _victimObject, getPlayerUID _victimObject, name _clientObject, getPlayerUID _clientObject, _moneyAmount, (_respectPercentage / 100), _package] call ExileServer_MostWanted_util_log;`

At least if it happens again, so long as the RPT doesn't get deleted, you can find that message. 
We won't know if the fix works until it happens again, so let me know.

**Comment by ReDBaroN1** [comment]
Yes, one player said his respect was zeroed but, I didn't check further into that. I guess if any of them had it increase a lot, they wouldn't tell me... ;)
Going ahead with the changes above with one slight change....I have logging switched off so, will modify your code above to infi custom log.
Big thanks again for your time on this

**Comment by ReDBaroN1** [comment]
Just promoted the fix and logging to all 3 of my live servers; Altis, Tanoa & Malden. Although in their heyday the former 2 were constantly full, they now barely exceed 15/20 players so, probably not going to help us much for testing. Also, the problem has only happened once on Altis anyway. 
Malden can fill in the evenings and it's on there that the problem happens more or less every time someone places a bounty, 75% of the time at least. So, we should know pretty quickly if the fix has worked/helped. I'll also ask some regular players and admins to set a few up. Will keep you posted.

**Comment by ReDBaroN1** [comment]
The good news is that we haven't had a re occurrence of this issue since making those changes and have had some bounties set without issue. I'm still keeping my eye on it though in case it happens again.

We are having a different issue though, sometimes completed contracts when claiming don't credit the reward. When you click it a second time, it says something about the amount being blank even though it definitely had an amount before (sorry didn't catch full message). Then on relog, the claimed contract has gone.

Any ideas or log ideas to try and track it down?
